src/api/app/models/repository.rb

```diff
@@ -2,6 +2,7 @@ class Repository < ApplicationRecord
   belongs_to :project, foreign_key: :db_project_id, inverse_of: :repositories
 
   before_destroy :cleanup_before_destroy
+  before_save :set_remote_project_name_to_empty_string
```

Comment:
Having this before_save callback means that we dont need to go through the entire app and check that remote_project_name is never set to nil. Otherwise we might break other parts of the app. It seems safer to me to just convert nil to an empty string by default?